jwst/flatfield/flat_field.py

```diff
@@ -419,17 +427,20 @@ def nirspec_fs_msa(output_model, f_flat_model, s_flat_model, d_flat_model, dispa
             flat_slits.append(slit_flat)
 
         # Now let's apply the correction to science data and error arrays.  Rely
-        # on array broadcasting to handle the cubes
+        # on array broadcasting to handle the cubes.
+        # Also update the variances using BASELINE algorithm.
+        flat_data_squared = slit_flat.data * slit_flat.data
         if not inverse:
             slit.data /= slit_flat.data
+            slit.var_poisson /= flat_data_squared
+            slit.var_rnoise /= flat_data_squared
+            slit.var_flat = (slit.data / slit_flat.data * slit_flat.err) ** 2
         else:
             slit.data *= slit_flat.data
-
-        # Update the variances using BASELINE algorithm
-        flat_data_squared = slit_flat.data ** 2
-        slit.var_poisson /= flat_data_squared
-        slit.var_rnoise /= flat_data_squared
-        slit.var_flat = (slit.data / slit_flat.data * slit_flat.err) ** 2
+            slit.var_poisson *= flat_data_squared
+            slit.var_rnoise *= flat_data_squared
+            slit.var_flat = (slit.data * slit_flat.data / slit_flat.err) ** 2
+            #np.place(slit.var_flat, np.isinf(slit.var_flat), np.nan)  # REMOVE?
```

Comment:
Note to self: See if np.place is still needed after another RT. If not, remove from code.